spswarehouse/general/selenium.py

```diff
@@ -132,52 +139,61 @@ def helper_get_multiple_elements_by_class_name(driver, class_name: str, wait_tim
 
 ### Select Text in Element
 
-def helper_select_visible_text_in_element_by_id(driver, element_id: str, text_to_select: str, wait_time_in_second=30):
+def helper_select_visible_text_in_element_by_id(driver, element_id: str, text_to_select: str, wait_time_in_seconds=30):
     """
     Waits for an element by ID and selects it by specified text.
     """
-    elem = WebDriverWait(driver, wait_time_in_second).until(EC.presence_of_element_located((By.ID, element_id)))
+    elem = _wait_for_element_to_be_present_and_return_it(driver, By.ID, element_id, wait_time_in_seconds)
     select = Select(elem)
     select.select_by_visible_text(text_to_select)
 
-def helper_select_visible_text_in_element_by_name(driver, element_name: str, text_to_select: str, wait_time_in_second=30):
+def helper_select_visible_text_in_element_by_name(driver, element_name: str, text_to_select: str, wait_time_in_seconds=30):
     """
     Waits for an element by name and selects it by specified text.
     """
-    elem = WebDriverWait(driver, wait_time_in_second).until(EC.presence_of_element_located((By.NAME, element_name)))
+    elem = _wait_for_element_to_be_present_and_return_it(driver, By.NAME, element_name, wait_time_in_seconds)
     select = Select(elem)
     select.select_by_visible_text(text_to_select)
 
 
 ### Type in Element
 
-def helper_type_in_element_by_id(driver, element_id: str, input_to_type: str, wait_time_in_second=30):
+def helper_type_in_element_by_id(driver, element_id: str, input_to_type: str, wait_time_in_seconds=30):
     """
     Waits for an element by ID, clears it, and types in the input.
     """
-    elem = WebDriverWait(driver, wait_time_in_second).until(EC.presence_of_element_located((By.ID, element_id)))
+    elem = _wait_for_element_to_be_present_and_return_it(driver, By.ID, element_id, wait_time_in_seconds)
     elem.clear()
     elem.send_keys(input_to_type)
 
-def helper_type_in_element_by_name(driver, element_name: str, input_to_type: str, wait_time_in_second=30):
+def helper_type_in_element_by_name(driver, element_name: str, input_to_type: str, wait_time_in_seconds=30):
     """
     Waits for an element by name, clears it, and types in the input.
     """
-    elem = WebDriverWait(driver, wait_time_in_second).until(EC.presence_of_element_located((By.NAME, 
-        element_name)))
+    elem =  _wait_for_element_to_be_present_and_return_it(driver, By.NAME, element_name, wait_time_in_seconds)
     elem.clear()
     elem.send_keys(input_to_type)
 
 
 ### Wait for Element
 
-def helper_wait_for_element_containing_specific_text(driver, expected_element_text, wait_time_in_second=30):
```

Comment:
I hesitate about the structure of these last two. At some point these functions are too specific, and whoever is trying to use these just needs to do the Selenium coding themselves.

Reply:
Yeah, I think you're right. I was trying to account for another tool that expected True/False to be returned, but now I can't find where that was. I'm turning this into just one function that crashes if the element can't be found. Also updating powerschool.py to use the new one.